core/storage/collection/gae_models.py

```diff
@@ -516,8 +513,7 @@ def _trusted_commit(
         snapshot_metadata_model.commit_cmds_user_ids = list(
             sorted(commit_cmds_user_ids))
 
-        snapshot_metadata_model.update_timestamps()
-        snapshot_metadata_model.put()
+        snapshot_metadata_model.put_depending_on_id(committer_id)
```

Comment:
Is it possible to ensure people call ``put_depending_on_id`` or one of the alternatives vs. ``put()`` itself? This seems like a potentially easy mistake to make.

Reply:
Yep, `put` (and other of its alternatives like `put_multi`) throw `NotImplementedError` for the models that inherit from `BaseHumanMaintainedModel`.

Reply:
Sounds good--thanks for clarifying.